test/e2e/e2e_test.go

```diff
@@ -769,6 +866,50 @@ func TestE2E(t *testing.T) {
 		}
 	})
 
+	// make sure that deleting service-cert-secret regenerates a working secret again
+	t.Run("serving-cert-secret-delete-data", func(t *testing.T) {
+		ns, err := createTestNamespace(adminClient, "test-"+randSeq(5))
+		if err != nil {
+			t.Fatalf("could not create test namespace: %v", err)
+		}
+		testServiceName := "test-service-" + randSeq(5)
+		testSecretName := "test-secret-" + randSeq(5)
+		labels := map[string]string{
+			"app": "test",
+		}
+		defer cleanupServiceSignerTestObjects(adminClient, testSecretName, testServiceName, ns.Name)
```

Comment:
Usually cleanup functions are called after successful creation rather than before.

Reply:
I copied it from the test above. anyways it need not be addresses as it uses metrics endpoint of service-ca-operator now.